src/history/hs_rec.c

```diff
@@ -760,6 +782,17 @@ __wt_hs_delete_key_from_ts(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint3
         ++hs_counter;
     }
 
+    /*
+     * Fail the eviction if we detect out of order timestamp when checkpoint is running. We cannot
+     * modify the history store to fix the out of order timestamp updates as it may make the history
+     * store checkpoint inconsistent.
+     */
+    if (checkpoint_running) {
+        ret = EBUSY;
```

Comment:
I'm a bit apprehensive that this change will introduce failure paths in reconciliation we haven't tested much before. This change deserves extensive format and MongoDB testing before merge.

Reply:
I have ran mongodb patch build and sys-perf patch build and the links are on the ticket.